src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -89,11 +98,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     if (request.getQueryString() != null) {
       redirectPath += "?" + request.getQueryString();
     }
-
+    String preferredSite = "";
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
       request.setAttribute(AUTH_URL, userService.createLogoutURL(redirectPath));
       request.setAttribute(AUTH_LABEL, "Logout");
+      preferredSite = getPreferredSite(userService.getCurrentUser().getUserId());
```

Comment:
preferredSite needs `toLowerCase()` here as respect to line 152

Reply:
getPreferredSite already returns the source in lowercase

Reply:
uh, right, i forgot that i've already set all the value in the select div to be lower case